### PR TITLE
HHH-14447 Add missing EventGroupListeners into FastSessionService and…

### PR DESCRIPTION
**Pull request by Sanne**
https://hibernate.atlassian.net/browse/HHH-14447

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/action/internal/EntityInsertAction.java [hunk 1]
```diff
@@ -263,13 +250,12 @@ protected boolean cacheAfterInsert(EntityDataAccess cache, Object ck) {
 
 	@Override
 	protected boolean hasPostCommitEventListeners() {
-		final EventListenerGroup<PostInsertEventListener> group = listenerGroup( EventType.POST_COMMIT_INSERT );
+		final EventListenerGroup<PostInsertEventListener> group = getFastSessionServices().eventListenerGroup_POST_COMMIT_INSERT;
 		for ( PostInsertEventListener listener : group.listeners() ) {
```
Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/action/internal/EntityInsertAction.java:254]: I think `getFastSessionServices().eventListenerGroup_POST_COMMIT_INSERT` can be `null`

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/action/internal/EntityInsertAction.java:254]: sorry looking at the code and it seems not

File: hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java [hunk 2]
```diff
@@ -166,47 +165,49 @@ protected boolean preDelete() {
...
+			eventListeners.fireLazyEventOnEachListener( this::newPostDeleteEvent, EntityDeleteAction::postCommitDeleteOnSuccess );
+		}
+		else {
+			eventListeners.fireLazyEventOnEachListener( this::newPostDeleteEvent, EntityDeleteAction::postCommitDeleteOnUnsuccessful );
+		}
+	}
+
+	private static void postCommitDeleteOnSuccess(PostDeleteEventListener listener, PostDeleteEvent event) {
+		if ( listener instanceof PostCommitDeleteEventListener ) {
+			listener.onPostDelete( event );
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java:196]: Missing cast?

**Review by beikov — CHANGES_REQUESTED** [review]
Looking good overall, there is just one small issue I think

File: hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java (hunk 2 above)
Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java:196]: hum yes that is looking odd. I suspect I can just remove the `if` - need to look better.

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java:196]: I think it should be something like
```
if ( listener instanceof PostCommitDeleteEventListener ) {
if ( success ) {

					listener.onPostDelete( event );
				}
				else {
					((PostCommitDeleteEventListener) listener).onPostDeleteCommitFailed( event );
				}
}
```

Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java:196]: I'll do the same I did for `PostInsertEventListener`

Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/action/internal/EntityDeleteAction.java]: pushed an additional commit

**Review by beikov — APPROVED** [review]

**Comment by dreab8** [comment]
Thanks @Sanne and @beikov , merged on master and 5.4

**Pull request closed without merging**